scRNA-seq/05-clustering_markers_scRNA.Rmd

```diff
@@ -0,0 +1,400 @@
+            NNGraphParam(k = 20, 
+                         type = "jaccard", 
+                         cluster.fun = "louvain"), 
+            plot_dim = "UMAP")
+```
+
+
+```{r clusterplot3, live = TRUE}
+clusterPlot(hodgkins_sce, 
+            HclustParam(cut.number = 10),
+            plot_dim = "UMAP")
+```
+
+
+## Identifying marker genes
+
+Assigning clusters is nice for visualization, but we would like to be able to move toward a biological interpretation of the clusters and identifying the cell types in each cluster.
+To that end, we can identify marker genes that are differentially expressed among clusters. 
+
+It is worth noting here that the statistical calculations here are more than a bit circular: we identified clusters first based on gene expression, then we are using those same clusters to find differences in gene expression. 
+The result is that even if there were no *true* clusters, we would always find marker genes! 
+For a much more technical exploration of this circularity (and a method to correct for it), see a preprint by [Gao et al. (2020)](https://arxiv.org/abs/2012.02936).
+In light of this, it is better to think about marker gene identification as an aid in interpreting the clustering results (and possibly extending insights to new data sets), rather than results that should be interpreted on their own, and we should be extremely wary of justifying cluster assignments solely based on these results!
+With that caveat, let's proceed.
+
+To identify marker genes, we will use the `scran::findMarkers()` function, which will rank genes by their differential expression by calculating pairwise statistics among clusters.
+We have a few options for how to determine the gene rankings and marker gene list for each cluster. 
+At one end could include genes that are differentially expressed in *any* pairwise comparison against our focal cluster, or at the other we could only include genes that are differentially expressed in *all* comparisons with that cluster. 
+We could also do something in between, including genes that differentiate the focal cluster from *some* fraction of the other clusters.
+For now, we will use the `findMarkers()` function to rank the genes in each cluster by their combined scores against *all* other clusters. 
+
+`findMarkers()` will return a list (technically a list-like object) of tables, one for each cell type, with statistics for each gene showing how well it differentiates that cell type against other types.
+
+
+```{r find_markers}
+markers <- scran::findMarkers(hodgkins_sce, 
+                              groups = hodgkins_sce$nncluster, 
+                              pval.type = "all")
+```
+
+Next we can look at one of those tables.
+We will start with the first cluster, which we will select from the list using the R standard double bracket `[[1]]` notation.
+We also doing a bit of transformation here to pull the gene name into a column of its own.
+
+```{r marker_table}
```

Comment:
Can we perhaps extend this idea (do everything we need to do for one element of the list like object "in isolation") to include saving a file? And then we could gloss over the use of `purrr` to the level of "Here's a link to the tutorial; we're going to do the same thing we did before but now to each element."

Reply:
I think this might work well. The question below about whether we really need to do all of the clusters is valid: Part of my concern was that we might get a very bad cluster as cluster 1, and we would not want to use that. So I wanted to be sure that you had all the options in the next step, so that you could hopefully find one with interpretable results!